pg_metadata/test/test_sql.py

```diff
@@ -36,3 +37,69 @@ def test_html_template(self):
                 "SELECT pgmetadata.get_dataset_item_html_content('pgmetadata','lines','main')")
         )
         self.assertEqual("<p>Test title</p><b>Test title</b>", result[0][0])
+
+    def test_trigger_calculate_fields(self):
+        """ Test if fields are correctly calculated on a layer having
+        a geometry """
+
+        # Adding data
+        dataset_feature = {
+            'table_name': "'lines'",
+            'schema_name': "'pgmetadata'",
+            'title': "'Test title'",
+            'abstract': "'Test abstract.'",
+        }
+        self._insert(dataset_feature, 'dataset')
+
+        # Test insert
+        sql = "SELECT geometry_type, projection_authid, spatial_extent FROM pgmetadata.dataset"
+        result = self.connection.executeSql(sql)
+        self.assertEqual(['LINESTRING', 'EPSG:4326', '3.854, 3.897, 43.5786, 43.622'], result[0])
+
+        # Test update
+        sql = "UPDATE pgmetadata.dataset SET title = 'test lines title' WHERE table_name = 'lines'"
+        self.connection.executeSql(sql)
+        sql = "SELECT title, geometry_type, projection_authid, spatial_extent FROM pgmetadata.dataset"
+        result = self.connection.executeSql(sql)
+        self.assertEqual(
+            ['test lines title', 'LINESTRING', 'EPSG:4326', '3.854, 3.897, 43.5786, 43.622'],
+            result[0]
+        )
+
+    def test_trigger_calculate_fields_without_geom(self):
+        """ Test the trigger on a geometry less table. """
```

Comment:
Je pensais à cette coupure, mais je l'avais pas demandé, car c'était l'autre coupure. ;-)
Gardons la maintenant que c'est fait, mais par contre il faut renommer avec mon autre suggestion.

```suggestion
    def test_trigger_calculate_fields_when_removing_geom(self):
        """ Test the trigger on an existing table where we remove the geometry. """
```